### PR TITLE
InheritorViolation was only being triggered on grand-childen classes

### LINKED ISSUE #10167: `@psalm-inheritors` does not catch error at class declaration
Took an example from the [documentation](https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-inheritors): https://psalm.dev/r/0d2892f51f

No issues are reported

### PR DESCRIPTION
**Pull request by robchett**
Fixes #10167

### CODE REVIEW

File: src/Psalm/Internal/Analyzer/ClassAnalyzer.php [hunk 1]
```diff
@@ -271,6 +272,22 @@ public function analyze(
             );
         }
 
+        $classUnion = new Union([new TNamedObject($fq_class_name)]);
```
Comment (weirdan) [inline src/Psalm/Internal/Analyzer/ClassAnalyzer.php:275]: we use snake case for variable names

File: src/Psalm/Internal/Analyzer/ClassAnalyzer.php [hunk 2]
```diff
@@ -271,6 +272,22 @@ public function analyze(
...
+        foreach ($storage->parent_classes + $storage->direct_class_interfaces as $parent_class) {
```
Comment (weirdan) [inline src/Psalm/Internal/Analyzer/ClassAnalyzer.php:276]: This makes me wonder whether the following should be flagged or not:

```php
/** @psalm-inheritors Whatever */
interface I {}
interface IC extends I {}
```

Comment (robchett) [reply src/Psalm/Internal/Analyzer/ClassAnalyzer.php:276]: I'm of the belief extending the interface should be explicitly stated if the annotation is used. Updated to analyse interfaces too.

**Review by weirdan — APPROVED** [review]

**Comment by weirdan** [comment]
Thanks!

**Pull request merged**